### PR TITLE
Only run tests when python files have changed

### PR DESCRIPTION
**Pull request by jkimbo**
## Description

This PR updates the GitHub tests workflow to only run if any Python files (or dependency changes) have changed.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by patrick91 — APPROVED** [review]
Looks good ^^ thank you!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -5,6 +5,11 @@ on:
     branches: [main]
   pull_request:
     branches: [main]
+    paths:
+      - "strawberry/**"
+      - "tests/**"
+      - "pyproject.toml"
+      - "poetry.lock"
```
Comment (patrick91) [inline .github/workflows/test.yml:12]: oh, shall we put the action itself so if we change it we also test it in the same PR?

Comment (jkimbo) [reply .github/workflows/test.yml:12]: Oh yeah good idea

**Comment by jkimbo** [comment]
Ah one problem with this @patrick91 is that we'll have to remove the branch protection rules around running tests before merging. Is that ok?

**Comment by patrick91** [comment]
> Ah one problem with this @patrick91 is that we'll have to remove the branch protection rules around running tests before merging. Is that ok?

yes, maybe in future we can add a new check that fails if we tests did run and fail, but doesn't fail if no tests did run. For now we're good I'd say :)

**Comment by jkimbo** [comment]
OK I've removed the protection rules.

**Comment by BryceBeagle** [comment]
Do you think it would have been better to keep the branch protection rule in-place, and then just skipped the _job_ not the workflow for specific paths?

**Comment by jkimbo** [comment]
@BryceBeagle We could do that but it would make the workflow more complicated. Since we require at least one person to review a PR before merging I don't think it's a problem.

**Pull request merged**